### PR TITLE
Head class

### PR DESCRIPTION
**Pull request by sssmc**
Head is feature complete and working based on limited testing with one and two servos.

### CODE REVIEW

**Comment by sssmc** [comment]
I think that this should be merged after PR #5

File: INMOOV-Head-Control/Head.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    float delayPerStep = 1 / updateRate;
+    int count = 1;
+
+    while(count <= (updateRate * duration)){                                    //Loop until count has reached number of steps needed
+        if(interruptOnSerial == 1 && Serial.available() > 0){                   //Exit loop if there is anything in the serial buffer
+            interrupted = true;
+            break;
+        }
+
+        if(*interruptFlag == true){                                             //Exit the loop if the interruptFlag is high
```
Comment (gordonso928) [inline INMOOV-Head-Control/Head.cpp:83]: Is there a reason why a pointer is used here but not at line 60?

**Review by gordonso928 — COMMENTED** [review]
Looks good, curious why a pointer is used in that way, see comment.

**Comment by sssmc** [comment]
> Looks good, curious why a pointer is used in that way, see comment.

Note sure if there is a better way of doing this, the if statement on line 60 is checking if the address the pointer is pointing to is 0 ie the default value if the user did not pass an address to the method. If there was no address passed to the method, the pointer is pointed to a local bool that is always false, meaning a interrupt will never be trigged, but if a address is passed to the method, then the address will not be zero and the value at that address will be checked for a interrupt(line 83)

**Review by gordonso928 — APPROVED** [review]
Looks good

**Comment by gordonso928** [comment]
> > Looks good, curious why a pointer is used in that way, see comment.
> 
> Note sure if there is a better way of doing this, the if statement on line 60 is checking if the address the pointer is pointing to is 0 ie the default value if the user did not pass an address to the method. If there was no address passed to the method, the pointer is pointed to a local bool that is always false, meaning a interrupt will never be trigged, but if a address is passed to the method, then the address will not be zero and the value at that address will be checked for a interrupt(line 83)

I am also not sure if there is a better way, the code works so I think this will be future optimization. We can optimize it later and maybe talk to instructors when we get back. I will merge the PR now.

**Pull request merged**